adapters/sovrn/sovrn.go

```diff
@@ -61,6 +39,69 @@ func (s *SovrnAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapt
 		}
 	}
 
+	errs := make([]error, 0, 1)
+	var err error
+	var validImps int = 0
+
+	for index, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: err.Error(),
+			})
+			continue
+		}
+
+		var sovrnExt openrtb_ext.ExtImpSovrn
+		if err := json.Unmarshal(bidderExt.Bidder, &sovrnExt); err != nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: err.Error(),
+			})
+			continue
+		}
+
+		tagId := getTagid(sovrnExt)
+		if tagId == "" {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "Missing required parameter 'tagid'",
+			})
+			continue
+		}
+
+		imp.TagID = getTagid(sovrnExt)
+
+		if imp.BidFloor == 0 && sovrnExt.BidFloor > 0 {
+			imp.BidFloor = sovrnExt.BidFloor
+		}
+
+		// Validate video params if appropriate
+		video := imp.Video
+		if video != nil {
+			if video.MIMEs == nil ||
+				video.MinDuration == 0 ||
+				video.MaxDuration == 0 ||
+				video.Protocols == nil {
+				errs = append(errs, &errortypes.BadInput{
+					Message: "Missing required video parameter",
+				})
+				continue
+			}
+		}
+
+		validImps++
+		request.Imp[index] = imp
+	}
+
+	if validImps == 0 {
+		return nil, errs
+	}
+
+	reqJSON, err := json.Marshal(request)
```

Comment:
At this point, this request can have both valid and invalid imps because on L92 you're just reassigning the valid imps to the same place in the imps array of the request. You'd actually have to remove the invalid ones as you find them in the for loop so that you're only left with the valid ones or a much easier way would be to just create a new `validImps` array and keep appending each valid imp you find to that array and then outside the for loop, you can just replace `request.Imp` with that new `validImps` array before JSON marshalling

Reply:
Good point. Done.